lib/date_range.rb

```diff
@@ -1,7 +1,46 @@
 require 'time'
+require_relative 'movie_reserver'
 
 module GreenBox
+  # When a movie is being rented you need to understand when that rental occurs.  To represent that interval of time, you will create a `DateRange` class.
   class DateRange
+    attr_reader :start_date , :end_date, :nights
 
+    def initialize(start_date, end_date)
+      @start_date = start_date
+      @end_date = end_date
+
+      if start_date > end_date
+        raise ArgumentError.new "The end date is before the start date"
+      end
+    end
+
+#  `contains(date)` - This method returns true if the date occurs on or after the start date and before the end date.
+    def contains(date)
+      if (date >= @start_date) && (date < @end_date)
+        return true
+      else
+        date < @start_date || date >= @end_date
```

Comment:
You don't need this 2nd conditional since if it's not inside you can return false.

```ruby
if (date >= @start_date) && (date < @end_date)
        return true
      else
        return false
end
```